lib/SRV/Volume/Remediated/Bundle.pm

```diff
@@ -0,0 +1,204 @@
+    my ( $env ) = @_;
+
+    my $C = $$env{'psgix.context'};
+    my $mdpItem = $C->get_object('MdpItem');
+    my $gId = $mdpItem->GetId();
+
+    my $remediated_item_id = $self->remediated_item_id;
+    my $itemFileSystemLocation =
+        Identifier::get_item_location($remediated_item_id);
+
+    my $remediatedMdpItem;
+    $remediatedMdpItem = MdpItem::EMMA->GetMdpItem($C, $remediated_item_id, $itemFileSystemLocation);
+
+    my @fileIds = $remediatedMdpItem->GetRemediatedFileIds();
+
+    if ( scalar @fileIds > 1 ) {
+        # return the zip
+        $self->content_type('application/zip');
+        $self->output_filename($remediatedMdpItem->Get('zipfile'));
+        $self->attachment_filename(basename($self->output_filename));
+        $self->is_extracted(0);
+    } else {
+        my $fileid = $fileIds[0];
+        $self->content_type($remediatedMdpItem->GetStoredFileMimeType($fileid));
+        my ( $fileName, $filePath ) = $remediatedMdpItem->GetFilePathMaybeExtract($fileid);
+        $self->attachment_filename($fileName);
+        $self->output_filename($filePath);
+        $self->is_extracted(1);
+    }
+
+}
+
+sub _get_response_headers {
+    my $self = shift;
+    my $headers = [ "Content-Type", $self->_content_type ];
+    my $filename = $self->attachment_filename;
+    my $disposition = qq{attachment; filename=$filename};
+
+    push @$headers, "Content-disposition", $disposition;
+    if ( defined $self->output_filename && -f $self->output_filename ) {
+        push @$headers, "Content-length", ( -s $self->output_filename );
+    }
+    return $headers;
+}
+
+sub _content_type {
+    my $self = shift;
+    return $self->content_type;
+}
+
+sub _action {
+    my $self = shift;
+    'remediated';
+}
+
+sub _fill_params {
+    my ( $self, $env, $args ) = @_;
+
+    my $C = $$env{'psgix.context'};
+    my $mdpItem = $C->get_object('MdpItem');
+
+    my $req = Plack::Request->new($env);
+
+    my %params = $self->_possible_params();
+
+    SRV::Utils::parse_env(\%params, [], $req, $args);
+
+    foreach my $param ( keys %params ) {
+        $self->$param($params{$param});
+    }
+
+    $self->id($mdpItem->GetId());
+    $self->handle(SRV::Utils::get_itemhandle($mdpItem));
+    $self->attachment(1);
+}
+
+sub _possible_params {
+    my $self = shift;
+
+    my %params = ();
+    $params{remediated_item_id} = undef;
+    $params{format} = undef;
+    $params{tracker} = undef;
+
+    return %params;
+}
+
+sub _authorize {
+    my $self = shift;
+    my $env = shift;
+
+    $self->restricted(0) unless ( SRV::Utils::under_server() );
+
+    unless ( defined $self->restricted ) {
+
+        my $C = $$env{'psgix.context'};
+        my $mdpItem = $C->get_object('MdpItem');
+        my $ar = $C->get_object('Access::Rights');
+        my $gId = $mdpItem->GetId();
+
+        my $final_access_status = $ar->assert_final_access_status($C, $gId);
+        my $download_access_status = $ar->get_remediated_items_access_status($C, $gId);
```

Comment:
@aelkiss this now uses a new method in `Access::Rights` that answers the remediated files access.